test/e2e/framework/registry/registry.go

```diff
@@ -127,6 +127,9 @@ func podManifest(podTestLabel string) (*v1.Pod, error) {
 		},
 	}
 	pod.Spec.Containers[0].SecurityContext.RunAsUser = ptr.To[int64](5123)
+	// setting HostNetwork to true so that the registry is accessible on localhost:<hostport>
+	// and we don't have to deal with any CNI quirks.
```

Comment:
This was kinda lazy shorthand, I assume portmap has been setup to only listen on singlestack or something like that, IMHO that's probably a bug but ...

Using hostNetwork ensures we can just bind directly to 5000 on all addresses on the host, so localhost should resolve.

Probably with rephrasing a bit and maybe a link back to the context.